executor/insert_common.go

```diff
@@ -642,6 +643,19 @@ func (e *InsertValues) fillRow(ctx context.Context, row []types.Datum, hasValue
 			}
 		}
 	}
+	tbl := e.Table.Meta()
+	// Handle exchange partition
+	if tbl.ExchangePartitionFlag {
+		is := e.ctx.GetDomainInfoSchema().(infoschema.InfoSchema)
+		pt, tableFound := is.TableByID(tbl.ExchangePartitionId)
+		if !tableFound {
+			return nil, errors.Errorf("exchange partition with table TableByID Failed")
+		}
+		err := pt.(table.PartitionedTable).CheckForExchangePartition(e.ctx, pt.Meta().Partition, row, tbl.ExchangePartitionDefId)
```

Comment:
Please add a check for the type assertion:

```
p, ok := pt.(table.PartitionedTable)
if !ok {
    ...
}
```
To avoid the panic in any case(bugs or some other unexpected cases) when we have a `pt` that is not a partition table.